app/src/main/java/com/example/playlistmaker/search/data/TrackRepositoryImpl.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.playlistmaker.search.data
+
+import com.example.playlistmaker.player.domain.Track
+import com.example.playlistmaker.search.domain.TracksRepository
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class TracksRepositoryImpl(private val networkClient: NetworkClient) : TracksRepository {
+    override fun searchTracks(expression: String): List<Track> {
+        try {
+            val response = networkClient.doRequest(TrackSearchRequest(expression))
+            if (response.resultCode != 200) {
+                return emptyList()
+            }
+            return (response as TrackResponse).results.map {
+                Track(
+                    it.trackName!!,
+                    it.artistName!!,
+                    SimpleDateFormat("mm:ss", Locale.getDefault()).format(it.trackTimeMillis),
+                    it.artworkUrl100!!,
+                    it.trackId!!,
+                    it.collectionName!!,
+                    it.releaseDate!!,
+                    it.primaryGenreName!!,
+                    it.country!!,
+                    it.previewUrl!!
+                )
+            }
+        } catch (error: Error) {
+            throw Exception(error)
+        }
```

Comment:
1) В catch лучше ловить `Throwable`:
```suggestion
        } catch (t: Throwable) {
            throw Exception(t)
        }
```
2. А зачем обрабатывать ошибку и всё равно выбрасывать эксепшн? Зачем тогда ловили ошибку если дальше выбрасываем